llvm/test/CodeGen/RISCV/rvv/subregister-undef-early-clobber.mir

```diff
@@ -14,9 +14,9 @@ body:             |
     ; CHECK-NEXT: [[INSERT_SUBREG:%[0-9]+]]:vrm4 = INSERT_SUBREG [[DEF]], [[PseudoVLE32_V_M1_]], %subreg.sub_vrm1_0
     ; CHECK-NEXT: dead $x0 = PseudoVSETIVLI 0, 210 /* e32, m4, ta, ma */, implicit-def $vl, implicit-def $vtype
     ; CHECK-NEXT: %pt2:vrm4 = IMPLICIT_DEF
-    ; CHECK-NEXT: [[INIT_UNDEF:%[0-9]+]]:vrm2 = INIT_UNDEF
+    ; CHECK-NEXT: [[INIT_UNDEF:%[0-9]+]]:vrm2nov0 = INIT_UNDEF
```

Comment:
Where are the nov0 register classes coming from in this test case? None of the instructions below are masked so I would have thought that `MRI->getRegClass(MO.getReg());` would have returned vrm2/vr anyway?

Reply:
`TRI->getSubRegisterClass(VRM4, sub_vrm2_0)` gives VRM2, but `TRI->getSubRegisterClass(VRM4, sub_vrm2_1)` is VRM2NoV0 for some reason.

Reply:
I found this weird as well, but I think the reason is that if we're allocating consecutive registers, then only the bottom one can be v0, so the rest end up with a nov0 regclass.

Reply:
Yeah, I guess it's trying to find the smallest register class. Which now that I think about it, a vrm2nov0 should be fine if it's always going to be inserted into the upper half of a vrm4, because as you say it can never be v0. So the RISC-V changes LGTM